akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -14,8 +14,15 @@ final case class HttpChallenge(scheme: String, realm: String,
 
   def render[R <: Rendering](r: R): r.type = {
     r ~~ scheme
-    if (realm != null) r ~~ " realm=" ~~#! realm
-    if (params.nonEmpty) params.foreach { case (k, v) ⇒ r ~~ ',' ~~ k ~~ '=' ~~# v }
+
+    if (realm != null) {
+      r ~~ " realm=" ~~#! realm
+      params.foreach { case (k, v) ⇒ r ~~ ',' ~~ k ~~ '=' ~~# v }
```

Comment:
A simpler solution could be to use `if (params.nonEmpty) r ~~ ","` instead of this line and then make the rest of the method unconditional.